### ISSUE TITLE
index.html referencing non-existing directory leads to very silent 'fail'

### ISSUE THREAD

**Issue by eddelbuettel** [rstudio/shiny-server#92]
We use shiny-server 1.2.3.368 on Ubuntu and are extremely happy with it.   

But we managed to stop it 'cold' simply by
- renaming an existing application directory `/srv/shiny-server/$someName`
- but _not_ updating the indx file `/srv/shiny-server/index.html` file
- now every other app is also taken down and just gives 'Not Found'
- despite multiple attempts at increasing verbosity in logs, we never managed to get an error message

So in short, this is a wishlist request to make this behavior both more robust, and possibly also bark a little more in the log files.
